python/pyproject.toml

```diff
@@ -0,0 +1,49 @@
+[build-system]
+requires = ["setuptools>=71.0", "setuptools_scm>=8"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "ccf"
+dynamic = ["version"]
```

Comment:
Can we add a comment explaining some of the magic behind this version selection, and/or pointing to the docs? I think the rest is reasonably self-explanatory.

Reply:
yes, done